packages/flutter/lib/src/material/stepper.dart

```diff
@@ -0,0 +1,648 @@
+    assert(this.currentStep != null);
+    assert(0 <= currentStep && currentStep < this.steps.length);
+  }
+
+  /// The steps of the stepper whose titles, subtitles, icons always get shown.
+  final List<Step> steps;
+
+  /// The type of stepper that determines the layout. In the case of
+  /// [StepperType.horizontal], the content of the current step is displayed
+  /// underneath as opposed to the [StepperType.vertical] case where it is
+  /// displayed in-between.
+  final StepperType type;
+
+  /// The index into [steps] of the current step whose content is displayed.
+  final int currentStep;
+
+  /// The callback called when a step is tapped, with its index passed as
+  /// an argument.
+  final ValueChanged<int> onStepTapped;
+
+  /// The callback called when the 'continue' button is tapped.
+  ///
+  /// If null, the 'continue' button will be disabled.
+  final VoidCallback onStepContinue;
+
+  /// The callback called when the 'cancel' button is tapped.
+  ///
+  /// If null, the 'cancel' button will be disabled.
+  final VoidCallback onStepCancel;
+
+  @override
+  _StepperState createState() => new _StepperState();
+}
+
+class _StepperState extends State<Stepper> {
+  List<GlobalKey> _keys;
+  final Map<int, StepState> _oldStates = new Map<int, StepState>();
+
+  @override
+  void initState() {
+    super.initState();
+    _keys = new List<GlobalKey>.generate(
+      config.steps.length,
+      (int i) => new GlobalKey()
+    );
+
+    for (int i = 0; i < config.steps.length; i += 1)
+      _oldStates[i] = config.steps[i].state;
+  }
+
+  @override
+  void didUpdateConfig(Stepper oldConfig) {
+    super.didUpdateConfig(oldConfig);
+    assert(config.steps.length == oldConfig.steps.length);
+
+    for (int i = 0; i < oldConfig.steps.length; i += 1)
+      _oldStates[i] = oldConfig.steps[i].state;
+  }
+
+  bool _first(int index) {
+    return index == 0;
+  }
+
+  bool _last(int index) {
+    return config.steps.length - 1 == index;
+  }
+
+  bool _current(int index) {
+    return config.currentStep == index;
+  }
+
+  Widget _line(bool visible) {
+    return new SizedBox(
+      width: visible ? 1.0 : 0.0,
+      height: 16.0,
+      child: new Container(
+        decoration: new BoxDecoration(
+          backgroundColor: Colors.grey[400]
+        )
+      )
+    );
+  }
+
+  Widget _circleChild(int index, bool oldState) {
+    StepState state = oldState ? _oldStates[index] : config.steps[index].state;
+    assert(state != null);
+    switch (state) {
+      case StepState.indexed:
+      case StepState.disabled:
+        return new Text('${index + 1}', style: _kStepStyle);
+      case StepState.editing:
+        return new Icon(Icons.edit, color: Colors.white);
+      case StepState.complete:
+        return new Icon(Icons.check, color: Colors.white);
+      case StepState.error:
+        return new Text('!', style: _kStepStyle);
+    }
+    return null;
+  }
+
+  Widget _buildCircle(int index, bool oldState) {
```

Comment:
some of your methods are called "_buildFoo" and some just "_foo". You should be consistent unless there's a clear reason for the distinction (in which case add a comment explaining it).